src/realm/uuid.hpp

```diff
@@ -0,0 +1,119 @@
+/*************************************************************************
+ *
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#ifndef REALM_UUID_HPP
+#define REALM_UUID_HPP
+
+#include <realm/string_data.hpp>
+
+namespace realm {
+
+struct InvalidUUIDString : std::logic_error {
+    InvalidUUIDString(std::string msg)
```

Comment:
You can just do `using std::logic_error::logic_error;` to import the constructor.